src/ray/raylet/scheduling_queue.cc

```diff
@@ -111,34 +107,48 @@ bool SchedulingQueue::TaskQueue::RemoveTask(const TaskID &task_id,
   return true;
 }
 
-bool SchedulingQueue::TaskQueue::HasTask(const TaskID &task_id) const {
+bool TaskQueue::HasTask(const TaskID &task_id) const {
   return task_map_.find(task_id) != task_map_.end();
 }
 
-const std::list<Task> &SchedulingQueue::TaskQueue::GetTasks() const { return task_list_; }
+const std::list<Task> &TaskQueue::GetTasks() const { return task_list_; }
 
-const ResourceSet &SchedulingQueue::TaskQueue::GetCurrentResourceLoad() const {
+const ResourceSet &TaskQueue::GetCurrentResourceLoad() const {
   return current_resource_load_;
 }
 
+bool ReadyQueue::AppendTask(const TaskID &task_id, const Task &task) {
+  auto &resources = task.GetTaskSpecification().GetRequiredResources();
+  RAY_CHECK(tasks_with_resources_[resources].insert(task_id).second == true);
```

Comment:
I think it's better for this to be a `std::deque` to maintain FIFO order within each resource set bucket.

Reply:
Also, this is a good use case where failure to perform an operation should not crash the raylet. Maybe return false instead of a RAY_CHECK?

Reply:
For the first comment, see above. For the second comment, I disagree. If the check fails, that's a bug in the code and crashing the raylet is fine in that case.